consensus/istanbul/backend/announce_task.go

```diff
@@ -0,0 +1,137 @@
+package backend
+
+import (
+	"time"
+
+	"github.com/celo-org/celo-blockchain/consensus/istanbul"
+	"github.com/celo-org/celo-blockchain/p2p"
+)
+
+// announceTaskState encapsulates the state needed to guide the behavior of the announce protocol
+// thread. This type is designed to be used from A SINGLE THREAD only.
+type announceTaskState struct {
```

Comment:
I like the ideas of modeling the protocol as a state machine. where you get "events" and react to them. Some events are "time", others are network messages, etc.

But seeing it like that, seems to make it simpler to understand

Reply:
still... not sure the separation of announceManager and announceTask state. My sense is that it explores this "state machine" modeling, but the announce manager is also the protocol...